### DISCUSSION TITLE
Multiple goal pose for robot

### DISCUSSION THREAD

**Discussion by dahmedm** [NVlabs/curobo#108] [Software Q&A] [answered]
Hi
When I run plan_goalset and give as as input two poses, it only plans for the first pose. What could be the issue? Also, can I slow down the speed of generated motion by changing any input argument?

**Comment by balakumar-s** [comment, marked as answer]
plan_goalset plans to reach one of the poses in the goalset. This is similar to equation 10 in https://arxiv.org/abs/2204.00134.

If you want to generate motion plans from a batch of start configs to a batch of goal poses, check plan_batch(). E.g., https://github.com/NVlabs/curobo/blob/c09d94908d8ad52f557ed59195c1ceb2d0434d65/examples/motion_gen_example.py#L250

To slow down motions, try a velocity_scale of 0.5. https://github.com/NVlabs/curobo/blob/c09d94908d8ad52f557ed59195c1ceb2d0434d65/src/curobo/wrap/reacher/motion_gen.py#L198
